### PR TITLE
diff-sequence 1.1.2 -- Convert DiffSequence.makeChangedFields Method to Utilize Map

### PR DESCRIPTION
**Pull request by mvogttech**
PR in response to [request](https://github.com/meteor/meteor/pull/11426#issuecomment-839971771) from @StorytellerCZ

### CODE REVIEW

**Review by StorytellerCZ — APPROVED** [review]

File: packages/diff-sequence/diff.js [hunk 1]
```diff
@@ -11,277 +16,256 @@ function isObjEmpty(obj) {
...
-//    if unordered, they are IdMaps
-DiffSequence.diffQueryChanges = function (ordered, oldResults, newResults,
-                                              observer, options) {
-  if (ordered)
+/**
+ * @function DiffSequence
+ * @description old_results and new_results: collections of documents. if ordered, they are arrays. if unordered, they are IdMaps
+ */
+
+export default DiffSequence = {
```
Comment (afrokick) [inline packages/diff-sequence/diff.js:24]: Why `export default`?

Maybe `export const DiffSequence = new class { };` ?

Comment (mvogttech) [reply packages/diff-sequence/diff.js]: You're right. A named export/import will be better for code refactoring in the future... which is the whole point of the PR.

The latest commit also includes a performance optimization to the isObjEmpty method. You can see the performance difference [here](https://jsbench.me/plkon2676h/1).

**Review by mvogttech — COMMENTED** [review]
I cannot understand why the TCI build is failing. Does anyone have any idea why?

`{"groupPath":["tinytest","livedata stub"],"test":"reconnect","events":[{"sequence":2310,"type":"fail","details":{"type":"assert_equal","expected":"{\"added\":2,\"removed\":0,\"changed\":3,\"moved\":0}","actual":"{\"added\":2,\"removed\":0,\"changed\":3,\"moved\":1}","not":false}}],"server":true}`

**Comment by StorytellerCZ** [comment]
@mvogttech If you look closely on the message, the issue is here:
Expected: `moved: 0`
Actual: `moved 1`

**Comment by mvogttech** [comment]
@StorytellerCZ -- Do you think this class should have a cache to memoize these iteration-intensive functions?

**Comment by filipenevola** [comment]
@mvogttech same here, it would be nice to have comments in the places where you are making actual changes.

**Comment by StorytellerCZ** [comment]
@mvogttech could you please fix the failing tests?

**Pull request closed without merging**